src/context.cpp

```diff
@@ -699,6 +700,40 @@ namespace librealsense
             software_sensor::open( profiles );
         }
 
+        void custom_on_video_frame( rs2_software_video_frame const & software_frame )
+        {
+            // We do exactly the same as the base on_video_frame(), but we have a custom releaser that know that the
+            // pixels are actually pointing to a vector:
+            auto p_pixels_vector = static_cast< std::vector< byte > * >( software_frame.pixels );
+            rsutils::deferred on_release( [p_pixels_vector]() { delete p_pixels_vector; } );
+
+            // This will all go away -- moving to storing frame_additional_data directly!
+            stream_profile_interface * profile = software_frame.profile->profile;
+            auto vid_profile = dynamic_cast<video_stream_profile_interface *>(profile);
+            if( ! vid_profile )
+                throw invalid_value_exception( "Non-video profile provided to on_video_frame" );
+
+            if( ! _is_streaming )
+                return;
+
+            frame_additional_data data;
+            data.timestamp = software_frame.timestamp;
+            data.timestamp_domain = software_frame.domain;
+            data.frame_number = software_frame.frame_number;
+            data.depth_units = software_frame.depth_units;
+
+            data.metadata_size = (uint32_t)( _metadata_map.size() * sizeof( rs2_metadata_type ) );
+            memcpy( data.metadata_blob.data(), _metadata_map.data(), data.metadata_size );
+
+            auto frame_i
```

Comment:
What does the `i` in `frame_i` stands for? `allocated_frame` or `new_frame` seems a more meaningful name

Reply:
it's a `frame_interface` (as opposed to a `frame`)

Reply:
I don't like it that the name is dependent on the type. The name should convey the purpose of the variable and should remain the same even if `allocate_new_video_frame` will change to return `frame` or `video_frame` and not `frame_interface`.